### PR TITLE
grueneisen parameter now modified by modify_parameters

### PR DESCRIPTION
**Pull request by bobmyhill**
The grueneisen parameter in burnman is a dependent property, and can therefore be modified by the property_modifiers. This PR fixes that issue. 

Sundman_1991.py is also updated in order to test this fix.

### CODE REVIEW

**Comment by tjhei** [comment]
built now:
```
##1       <== BURNMAN/burnman/mineral.py: RuntimeWarning: invalid value encountered in double_scalars
          ==>
```

**Comment by bobmyhill** [comment]
Thanks @tjhei 
I don't think there's a good solution to this problem. We could 
 - calculate at some very low T if T < delta K, 
 - estimate at zero using l'Hopital's rule (but using numerical differentiation, so very unlikely to be better than just calculating grueneisen directly at low T), or 
 - raise an error if T < delta K.

Thoughts?

**Comment by tjhei** [comment]
Sorry, what is the cause for the error? Is the test sensible? Is the current output sensible?

If the output is not sensible, raising an error would make the most sense, I think.

**Comment by bobmyhill** [comment]
The error is related to how the grueneisen parameter is defined at 0 K. 

\gamma is identically equal to \alpha K_T V / C_v, and defining it this way rather than in every equation of state guarantees consistency of the equation of state when using property modifiers (which apply a free energy modifier as a function of P and T).

Unfortunately, at 0 K, \alpha and C_v are both identically zero, hence the divide by zero error even when grueneisen may have a well-defined non-zero value. 

So, in the old formulation, grueneisen was correct only if the property_modifiers didn't modify the EoS value. In the new formulation, grueneisen is always correct at finite temperatures, but undefined at 0 K.

**Comment by trg818** [comment]
Two brief thoughts, as you asked for it:
I may be mistaken, but I think the Grüneisen parameter is not supposed to be strongly T-dependent; if that is true, it may be sufficient to calculate it at a "safe" distance from 0 K where the quotient doesn't yet blow up.
A problem with de l'Hôpital's rule is that the first derivatives of alpha and cV are also zero at 0 K. So with the relation you quote, you would have to go at least to the 2nd derivative.

**Comment by bobmyhill** [comment]
Hi Thomas, thanks for your comments. Yes, any form of differentiation could be very problematic ... The order would depend on the equation of state (many don't have the correct limiting behaviour at 0 K).

I'm in favour of adding a try/except with an explicit note to try to recalculate at some small T. But open to alternatives.

**Comment by bobmyhill** [comment]
One alternative... We could use the EOS defined grueneisen if present and if there are no modifiers, otherwise try alpha KT V/Cv, with an except statement if calculation fails.

**Comment by bobmyhill** [comment]
Thanks everyone for your help with this. I've found a solution that retains old functionality, fixes the problem, and will raise an error if the user does something unexpected (i.e. defining gibbs modifiers, not defining a grueneisen parameter in their EoS *and* wanting to return grueneisen at 0 K).

File: tests/test_eos_consistency.py [hunk 1]
```diff
@@ -26,6 +26,14 @@ def test_SLB(self):
         T = 3000.
         self.assertEqual(burnman.tools.check_eos_consistency(burnman.minerals.SLB_2011.periclase(), P, T),
                          True)
+        
+    def test_modifier(self):
+        P = 10.e9
+        T = 3000.
+        with warnings.catch_warnings(record=True) as w:
+            warnings.simplefilter("always")
```
[the remark this answers is not in the document]
Comment (bobmyhill) [reply tests/test_eos_consistency.py]: It generated the shear properties warning. However, with PR #346 that @sannecottaar merged today, I can turn this warning off. The new commit does exactly that.

**Comment by tjhei** [comment]
looks reasonable!

**Pull request merged**